avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1492,13 +1511,11 @@ private Schema toConnectSchema(org.apache.avro.Schema schema,
                   + " property must be a JSON Integer."
                   + " https://avro.apache.org/docs/1.7.7/spec.html#Decimal");
             }
+            // Capture the precision as a parameter only if it is not the default
             Integer precision = precisionNode.asInt();
-            builder.parameter(CONNECT_AVRO_DECIMAL_PRECISION_PROP, precision.toString());
-          } else {
-            builder.parameter(
-                CONNECT_AVRO_DECIMAL_PRECISION_PROP,
-                CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT.toString()
-            );
+            if (precision != CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT) {
```

Comment:
What happens when someone uses the default precision explicitly?

Reply:
It still works.

When creating a Connect schema programmatically, Connect doesn't set this as a property but will generate an Avro schema with the precision (that code is elsewhere). However, when recovering a Connect schema from an Avro schema, this code always added this precision property to the Connect schema properties.

This causes difficulty in the tests when a) creating a Connect schema, converting that to Avro, converting the Avro schema back to Connect, and then comparing the "before" and "after" Connect schema using `Schema.equals()`. This check would fail because the "after" Connect schema had the precision property, but the "before" never does. The two schemas are otherwise identical.

This change will now only add the precision property to the Connect schema if the value is not equal to the default. In the example, the "before" schema always uses a default precision of 64, so the "after" Connect schema will now never have precision property, and `Schema.equals()` will be correct. IF an Avro schema is created with a precision of something other than 64, this will be included as a property in the Connect schema, so that if that schema is used to generate an Avro schema the correct precision will be kept and used.